src/wallet/rpcwallet.cpp

```diff
@@ -3118,25 +3123,25 @@ UniValue z_getaddressforaccount(const UniValue& params, bool fHelp)
     }
     libzcash::AccountId account = accountInt;
 
-    std::set<libzcash::ReceiverType> receivers;
+    std::set<libzcash::ReceiverType> receiverTypes;
     if (params.size() >= 2) {
-        const auto& pools = params[1].get_array();
-        for (unsigned int i = 0; i < pools.size(); i++) {
-            const std::string& p = pools[i].get_str();
+        const auto& parsed = params[1].get_array();
+        for (unsigned int i = 0; i < parsed.size(); i++) {
```

Comment:
nit, as long as you're touching this line
```suggestion
        for (size_t i = 0; i < parsed.size(); i++) {
```